### ISSUE TITLE
LTO support

### ISSUE THREAD

**Issue by bircoph** [clsync/clsync#173]
Support -flto whenever applicable.

**Comment by xaionaro** [comment]
@bircoph :
It does not compile on my machine anymore:
```
./configure: line 10300: syntax error near unexpected token `$LTOFLAGS,'
./configure: line 10300: `              AX_CHECK_LINK_FLAG($LTOFLAGS, have_lto_link=1)'
```

**Comment by bircoph** [comment]
Fixed in commit 88110330a5251f66a852cb32e6efcf99a841ba1c
Travis also catched this issue.

The problem is following: on most systems AX_CHECK_LINK_FLAG is undefined, only AX_CHECK_COMPILE_FLAG is defined. I have no idea why, autoconf has both these flags documented. The original code was tested on Gentoo where all AX_CHECK_*_FLAG's are present, though it looks like at least on Ubuntu (travis), Alt and your system the LINK flag is absent.

So I relaxed test to check only for COMPILE flag. It is less precise but should be sufficient.
